plugins/chat/lib/chat_sdk/message.rb

```diff
@@ -40,6 +40,73 @@ def self.create_with_stream(**params, &block)
     end
 
+    # Streams to a specific chat message.
+    #
+    # @param raw [String] text to append to the existing message.
+    # @param message_id [Integer] the ID of the message to stream.
+    # @param guardian [Guardian] an instance of the guardian class, representing the user's permissions.
+    # @return [Chat::Message] The message object.
+    # @example Streaming a message
+    #   ChatSDK::Message.stream(message_id: 42, guardian: guardian, raw: "text")
+    def self.stream(raw:, message_id:, guardian:, &block)
+      new.stream(raw: raw, message_id: message_id, guardian: guardian, &block)
+    end
+
+    # Starts streaming for a specific chat message.
+    #
+    # @param message_id [Integer] the ID of the message for which streaming should be stopped.
+    # @param guardian [Guardian] an instance of the guardian class, representing the user's permissions.
+    # @return [Chat::Message] The message object.
+    # @example Starting the streaming of a message
+    #   ChatSDK::Message.start_stream(message_id: 42, guardian: guardian)
+    def self.start_stream(message_id:, guardian:)
+      new.start_stream(message_id: message_id, guardian: guardian)
+    end
+
+    # Stops streaming for a specific chat message.
+    #
+    # @param message_id [Integer] the ID of the message for which streaming should be stopped.
+    # @param guardian [Guardian] an instance of the guardian class, representing the user's permissions.
+    # @return [Chat::Message] The message object.
+    # @example Stopping the streaming of a message
+    #   ChatSDK::Message.stop_stream(message_id: 42, guardian: guardian)
+    def self.stop_stream(message_id:, guardian:)
+      new.stop_stream(message_id: message_id, guardian: guardian)
+    end
+
+    def start_stream(message_id:, guardian:)
+      message = Chat::Message.find(message_id)
+      guardian.ensure_can_edit_chat!(message)
+      message.update!(streaming: true)
+      ::Chat::Publisher.publish_edit!(message.chat_channel, message.reload)
+      message
+    end
+
+    def stream(message_id:, raw:, guardian:, &block)
+      message = Chat::Message.find(message_id)
+      helper = StreamHelper.new(message, guardian)
+      helper.stream(raw: raw)
+      ::Chat::Publisher.publish_edit!(message.chat_channel, message.reload)
+      message
+    end
+
+    def stop_stream(message_id:, guardian:)
+      with_service(Chat::StopMessageStreaming, message_id: message_id, guardian: guardian) do
+        on_success { result.message }
+        on_model_not_found(:message) { raise "Couldn't find message with id: `#{message_id}`" }
+        on_failed_policy(:can_join_channel) do
+          raise "User with id: `#{guardian.user.id}` can't join this channel"
+        end
+        on_failed_policy(:can_stop_streaming) do
+          raise "User with id: `#{guardian.user.id}` can't stop streaming this message"
+        end
+        on_failure do
+          p Chat::StepsInspector.new(result)
```

Comment:
I would probably only do this kind of printing in Rails.env.dev? otherwise it ends up in weird places